File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.cpp
class Solution {
public:
    long long minimumSteps(string s) {
        priority_queue<int> pq;
        long long ans = 0;
        
        for(int i=s.size()-1;i>=0;i--){
            if(s[i] == '0'){
                pq.push(i);
            }
            else if(s[i] == '1'){
                if(!pq.empty()){
                    ans += pq.top() - i;
                    pq.pop();
                    pq.push(i);
                }
            }
        }
        return ans;
    }
};

File: 2938-separate-black-and-white-balls/2938-separate-black-and-white-balls.java
class Solution {
    public long minimumSteps(String s) {
        int low = 0;
        int high = s.length()-1;
        
        long ans = 0L;
        while(low <= high){
            if(s.charAt(low) == '1' && s.charAt(high) == '0'){
                ans += (high-low);
                low++;
                high--;
            }
            else if(s.charAt(low) == '0'){
                low++;
            }
            else if(s.charAt(high) == '1'){
                high--;
            }
        }
        return ans;
    }
}